internal/services/netapp/netapp_volume_quota_rule_data_source.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package netapp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumequotarules"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	netAppModels "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/models"
+	netAppValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type NetAppVolumeQuotaRuleDataSource struct{}
+
+var _ sdk.DataSource = NetAppVolumeQuotaRuleDataSource{}
+
+func (r NetAppVolumeQuotaRuleDataSource) ResourceType() string {
+	return "azurerm_netapp_volume_quota_rule"
+}
+
+func (r NetAppVolumeQuotaRuleDataSource) ModelObject() interface{} {
+	return &netAppModels.NetAppVolumeQuotaRuleDataSourceModel{}
+}
+
+func (r NetAppVolumeQuotaRuleDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumequotarules.ValidateVolumeQuotaRuleID
+}
+
+func (r NetAppVolumeQuotaRuleDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+
+		"volume_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeID,
```

Comment:
`volumequotarules` actually has a validate volume id function already that we can use instead
```suggestion
			ValidateFunc: volumequotarules.ValidateVolumeID,
```

Reply:
Ah, good catch, I wasn't aware of it.